src/css.js

```diff
@@ -263,8 +264,12 @@ jQuery.extend( {
 			// If a hook was provided, use that value, otherwise just set the specified value
 			if ( !hooks || !( "set" in hooks ) ||
 				( value = hooks.set( elem, value, extra ) ) !== undefined ) {
+				if ( typeof value === "string" && value.indexOf( "!important" ) !== -1 ) {
 
-				if ( isCustomProp ) {
+					// Support !important priority (gh-3713).
+					var splitValue = value.split( "!" );
+					style.setProperty( argName,  splitValue[ 0 ], splitValue[ 1 ] );
```

Comment:
This will not work for properties that need to be prefixed as you're using the unprocessed `argName` variable.